fs/vfs/fs_fsync.c

```diff
@@ -53,23 +54,31 @@
 
 int file_fsync(FAR struct file *filep)
 {
-  struct inode *inode;
+  FAR struct inode *inode;
+  int ret;
 
   /* Is this inode a registered mountpoint? Does it support the
    * sync operations may be relevant to device drivers but only
    * the mountpoint operations vtable contains a sync method.
    */
 
   inode = filep->f_inode;
-  if (!inode || !INODE_IS_MOUNTPT(inode) ||
-      !inode->u.i_mops || !inode->u.i_mops->sync)
+  if (inode != NULL)
     {
-      return -EINVAL;
+      if (INODE_IS_MOUNTPT(inode) && inode->u.i_mops->sync)
+        {
+          /* Yes, then tell the mountpoint to sync this file */
+
+          return inode->u.i_mops->sync(filep);
+        }
+      else if (inode->u.i_ops->ioctl)
```

Comment:
Before checking `inode->u.i_ops->ioctl` I think we need to check `inode->u.i_ops` is not NULL, for same reason as above.

Reply:
inode->u.i_mops

Reply:
Done. Thank you~